lesson21/src/js/main.js

```diff
@@ -0,0 +1,212 @@
+import '../css/style.css'
+import * as loading from "./module/loading";
+import { createAttributedElements} from "./utils/createAttributedElements";
+
+const endPointURL = {
+    usersData: "https://myjson.dit.upm.es/api/bins/i55y"
+}
+
+const createErrorMessage = (error,element) => {
+    const errorMessage = createAttributedElements({
+        tag:"p",
+        valuesByAttributes:{
+            class:"error-message"
+        },
+        str:error
+    });
+    element.appendChild(errorMessage);
+}
+
+const fetchData = async (endpointURL,element) => {
+    const response = await fetch(endpointURL);
+    if(!response.ok){
+        console.error(`${response.status}:${response.statusText}`);
+        createErrorMessage("Communication with the server is broken.",element);
+        return;
+    }
+    const json = await response.json();
+    return json.data;
+}
```

Comment:
You are right, the process of extracting the value (data) from the json is a better way to do it, if it is done in initUsersData.

This is because by clearly separating the role of fetch, it is easier to identify errors and bugs.

I fixed!
https://github.com/nakagawamai/js-lesson/pull/23/commits/839b5b35dfce88d54fba973193b4a4e96467eb97

Reply:
Thank you for the correction and feedback.

nits
It can be written this way.

```
const json = await response.json();
return json;

↓

return await response.json();
```

Reply:
Indeed it is!
I fixed.
https://github.com/nakagawamai/js-lesson/pull/23/commits/f022c4494d9f5746f4ee22722fe206e52e53fed9